cmd/app/serve.go

```diff
@@ -85,6 +100,7 @@ func newServeCmd() *cobra.Command {
 	cmd.Flags().String("grpc-port", "8081", "The port on which to serve requests for GRPC")
 	cmd.Flags().String("metrics-port", "2112", "The port on which to serve prometheus metrics endpoint")
 	cmd.Flags().Duration("read-header-timeout", 10*time.Second, "The time allowed to read the headers of the requests in seconds")
+	cmd.Flags().Bool("duplex", false, "experimental: serve HTTP and GRPC on the same port instead of on two separate ports")
```

Comment:
Should we also mark grpc-host/grpc-port deprecated in their flag descriptions? Then maybe later on we can use something like:
https://pkg.go.dev/go.etcd.io/etcd/pkg/flags#DeprecatedFlag

Reply:
Added! I also added in a warning to suggest to users that they switch to using the --duplex flag if it isn't set.

Reply:
Thanks!, that's a great idea. wonder if we should do something about the http-host / http-port aliases as well, since with duplex they are kind of wonky? Up to you, but now that I think about it, in the future they too should be deprecated, removed and only use host/port?

Reply:
I think it makes sense to remove the aliases! They were added in back in April (https://github.com/sigstore/fulcio/pull/472) so I think it's been enough time since then. I guess whenever we actually remove grpc-port grpc-host we can remove http-port and http-host as well.